src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs

```diff
@@ -61,18 +62,22 @@ public class AzureVirtualNetworkSubnetConfigBase : NetworkBaseCmdlet
             HelpMessage = "RouteTable")]
         public PSRouteTable RouteTable { get; set; }
 
+        [GenericBreakingChange("Update Property Name", OldWay = "-ResourceId", NewWay = "-NatGatewayId")]
```

Comment:
We shouldn't have a breaking change attribute as we won't remove ResourceId since it is a standard parameter for Azure PowerShell cmdlets.

Reply:
this was agreed upon in offline meeting with @niander  and @markcowl. We will retain resourceid on natgateway operations but can change parameter name when it is referenced from another operation. (Similar to RouteTableId here)

Reply:
Discussed with Niander. Looks good.